module/os/freebsd/zfs/zvol_os.c

```diff
@@ -292,6 +292,7 @@ zvol_geom_open(struct g_provider *pp, int flag, int count)
 			if (!mutex_tryenter(&spa_namespace_lock)) {
 				mutex_exit(&zv->zv_state_lock);
 				rw_exit(&zv->zv_suspend_lock);
+				drop_suspend = TRUE;
```

Comment:
```suggestion
				drop_suspend = B_FALSE;
```
You seem to be right that assignment missing, except it should be B_FALSE, since we just dropped the lock.